doc/design/secret-provisioning.md

```diff
@@ -0,0 +1,167 @@
+# Flexible Secret provisioning with Mystikos
+
+For confidential computing applications, we don't recommend that the
+application code to contain any secrets, as the code (either in source
+or binary forms) could be exposed to hostile environment during
+downloading, uploading, copying, or launching of the application.
+Instead, we recommend the secrets to be securely released to the
+application while it is executed in a Trused Execution Environment (TEE).
+The providers of the secrets could be a key vault service such as AKV,
+M-HSM, or a service run by the user in a locked down and trusted
+environment.
+
+This design document describes the process of secret provisioning
+ with Mystikos in 5 aspects:
+
+1. **the user interface**: how does Mystikos allow users to specify the
+secrets they want in their applications?
+1. **the client library**: how does Mystikos provide a reference secret
+provisioning client library and how can users plug in an alternative library.
+1. **the server library**: how does Mystikos provide a reference secret
+provisioning server library and how can users plug in an alternative library.
+1. **the secret release service**: how does the secret holder use the
+server library for secure release of secrets.
+1. **the attestation model**: the evidences and the verification of
+such evidences agreed by both the client and the server.
+
+## The user interface
+
+Users who wish to provision secrets to their applications running with
+Mystikos can add the following section to config.json:
+
+```json
+secrets: [
+    # Secret 1
+    {
+        "Name": <name of the secret>,
+        "AttestationModel": <a number between 1 to 3. See attestation models below>
+        "HolderURL": <URL to the secret release service>,
+        "Kid": <identifier of the key>,
```

Comment:
I suggest "KeyId" instead of "Kid", which sounds like a child.

Reply:
`Kid` is used in M-HSM documents like [here](https://docs.microsoft.com/en-us/azure/key-vault/managed-hsm/hsm-protected-keys-byok). But I agree, `KeyID` is a better name.